src/scenechange/mod.rs

```diff
@@ -200,7 +199,9 @@ impl<T: Pixel> SceneChangeDetector<T> {
 
     if scenecut {
       // Clear buffers and deque
-      self.frame_buffer.clear();
+      if let Some((_, is_initialized)) = &mut self.frame_buffer {
+        *is_initialized = false;
+      }
```

Comment:
This issue has preceded this branch, but I think we also need to clear the frame buffer and score deque on line 147, where we check for max keyframe interval:

```
if distance >= self.encoder_config.max_key_frame_interval {
  return true;
}
```

Since that block returns immediately, I think it could cause some odd behavior if the max scenecut distance is reached because it won't clear the buffer and score queue.

Reply:
I have added the buffer and score deque clears to the max key frame interval check.